Add Login and Register Pages

ludwigschubi opened the pull request:
## Related Ticket/Issue
fixes #6

## Description
This PR adds:
a Login Page that links our Login and Register Pages, as well as the ones from other Identity providers

## Related PR's
Describe related Pull Requests here:
-

## Added dependencies
Describe potential dependencies that were added with this PR:
-

bejow commented on src/actions/types.js:
```diff
@@ -33,3 +33,7 @@ export const SET_SELECTION = 'set_selection';
 export const SEND_NOTIFICATION_SUCCESS = 'send_notification_success';
 export const SEND_NOTIFICATION_FAILURE = 'send_notification_failure';
+
+export const FETCH_IDPS = 'fetch.idps';
```
@ludwigschubi 
let's stick to one naming convention words.with.dots or words_with_underscore

bejow commented on src/functional_components/IdentityProvider/package.json:
```diff
@@ -0,0 +1 @@
+{"main": "IdentityProvider.js"}
```
@ludwigschubi Why do you use package.json instead of an export?

ludwigschubi replied:
It's my personal preference to use package.json, but I don't have a specific reason for it. I guess we should just decide on one way of doing it.

bejow commented on src/functional_components/Navigation/Navigation.js:
```diff
@@ -61,12 +61,7 @@ const Navigation = ({ picture, webId, onLogin, onLogout, toggleSidebar }) => {
-                                            onLogin();
-                                        }}
-                                    >
+                                    <a href="http://localhost:3000/login">
```
Very unstable, would fail in production. Use react_router's Link instead of an <a> with hardcoded url

bejow commented on src/stateful_components/LoginScreen/LoginScreen.js:
```diff
@@ -1,21 +1,62 @@
+
+    onLogin(loginUrl) {
+        const { login } = this.props;
+        localStorage.setItem('returnToUrl', 'http://localhost:3000/home');
```
@ludwigschubi Use redux state for global variables, otherwise it gets confusing where stuff is saved. Also hardcoded urls are tricky, this can lead to bugs later on (port can change, domain can change etc.)

ludwigschubi replied:
This was meant to be used from solid-auth-client's-'Return to Origin' mechanism. But I already fixed it differently so it can actually be deleted.

bejow replied:
Ok, one idea would be to store window.location.href

bejow commented on src/utils/fileUtils.js:
```diff
@@ -233,22 +233,18 @@ function getFolderContents(folderUrl) {
+        console.log(
+            store.each(rdf.sym(folderUrl), ns.ldp('contains'), undefined)
+        );
+        console.log('DEBUG');
```
@ludwigschubi remove console logs before pull request --> keep master clean

ludwigschubi commented:
Okay I think i fixed most of the things you, rightly, pointed out. I think it's ready to be merged now. I also had to delete some things in other files for eslint to not throw errors, just so you know.

ludwigschubi commented:
Would still be good if you could test it one more time. May be that I forgot something.

bejow commented on src/functional_components/LoginForm/LoginForm.js:
```diff
@@ -0,0 +1,56 @@
+                />
+                {idps
+                    ? idps.map((idp, index) => {
+                          switch (idp.title) {
```
@ludwigschubi Could be extracted to a separte `getIdpStyle(idp.title)` function

bejow replied:
@ludwigschubi 
This branch is 8 commits behind master, can you pull the master and extract this function? Then I would say its ready for merging

bejow commented:
@ludwigschubi 
Seems like their icon is down or moved, what do you think of a fallback image for such cases? Do you have another idea?

![image](https://user-images.githubusercontent.com/42886506/64124506-7c4a7e00-cda7-11e9-815b-59a5a4ca771b.png)

bejow commented:
@ludwigschubi
I tested the login process couple of times because I got an "unauthorized" after I selected owntech as my idp. Now it seems to work like charm but I wonder why this happened the first 2 or 3 tries.

As you said, I think we should merge it anyway for the presentation and our other branches/tickets. If you have time till wednesday, the idp image alternative, if no image can be fetched, would be nice for the presentation

ludwigschubi commented:
> @ludwigschubi
> Seems like their icon is down or moved, what do you think of a fallback image for such cases? Do you have another idea?
> 

I removed the description for now since this made the button so large. I also removed the emojis from the Login page. I guess it's best if we keep this PR small so we can easily extend the Login Page in the future.
